python/lsst/ts/FiberSpectrograph/avsFiberSpectrograph.py

```diff
@@ -192,11 +198,18 @@ def _connect(self, serial_number=None):
         self.log.info("Activated connection (handle=%s) with USB device %s.", self.handle, device)
         self.device = device
 
+        # store the number of pixels for use when taking exposures.
+        n_pixels = _getUShortPointer()
+        code = self.libavs.AVS_GetNumPixels(self.handle, n_pixels)
+        assert_avs_code(code, "GetNumPixels")
+        self._n_pixels = n_pixels.contents.value
+
     def disconnect(self):
         """Close the connection with the connected USB spectrograph.
         If the attempt to disconnect fails, log an error messages.
         """
         if self.handle is not None and self.handle != AvsReturnCode.invalidHandle:
+            self.stop_exposure()  # stop any active exposure
```

Comment:
There are only two such uses: one in `disconnect` that has to deal with the possibility that we are cleaning up after a failed `connect` (and thus it could be None), and the one in `connect` that only has to deal with the `invalidHandle` return code since it just called `Activate`. I think those are different enough, and because of RAII we don't have any places that would need it, so I don't think we want that.

Reply:
Sounds good. I figured you'd have it anywhere you were about to try to use the handle, but didn't read the code again to check if that was so.

Reply:
Since this is RAII, if the handle is invalid outside of those two cases, everything is broken and the only way to fix it is to delete the object and make a new one.